pkgs/applications/graphics/processing/default.nix

```diff
@@ -1,73 +1,94 @@
-    rm core/library/*.jar
-    cp ${javaPackages.jogl_2_3_2}/share/java/*.jar core/library/
-
-    # do not download a file during build
-    ${xmlstarlet}/bin/xmlstarlet ed --inplace -P -d '//get[@src="http://download.processing.org/reference.zip"]' build/build.xml
-    install -D -m0444 ${fetchurl {
-                          # Use archive.org link for reproducibility until the following issue is fixed:
-                          # https://github.com/processing/processing/issues/5711
-                          url = "https://web.archive.org/web/20200406132357/https://download.processing.org/reference.zip";
-                          sha256 = "093hc7kc9wfxqgf5dzfmfp68pbsy8x647cj0a25vgjm1swi61zbi";
-                        }
-                       } ./java/reference.zip
-
-    # suppress "Not fond of this Java VM" message box
-    substituteInPlace app/src/processing/app/platform/LinuxPlatform.java \
-      --replace 'Messages.showWarning' 'if (false) Messages.showWarning'
-
-    ( cd build
-      substituteInPlace build.xml --replace "jre-download," ""  # do not download jre1.8.0_144
-      mkdir -p linux/jre1.8.0_144                               # fake dir to avoid error
-      ant build )
+    tar --checkpoint=10000 -czf build/linux/jdk-11.0.12.tgz ${jdk11}
+    cp ${ant}/lib/ant/lib/{ant.jar,ant-launcher.jar} app/lib/
+    mkdir -p core/library
+    cp ${javaPackages.jogl_2_3_2}/share/java/* core/library/
+    cp ${vaqua} app/lib/VAqua9.jar
+    unzip -qo ${jna} -d app/lib/
+    mv app/lib/{jna-5.8.0/dist/jna.jar,}
+    mv app/lib/{jna-5.8.0/dist/jna-platform.jar,}
+    GLUON="java/libraries/javafx/library"
+    mkdir -p $GLUON/linux64/modules/
+    mkdir -p $GLUON/macosx/modules/
+    unzip -qo ${gluon} -d $GLUON/
+    cp $GLUON/javafx-sdk-16/lib/*.jar $GLUON/macosx/modules/
+    cp $GLUON/javafx-sdk-16/lib/*.so $GLUON/linux64/
+    unzip -qo ${batik} -d java/libraries/svg/library/
+    cp java/libraries/svg/library/{batik-1.14/lib/,}batik-all-1.14.jar
+    cp ${ffmpeg} build/shared/tools/MovieMaker/ffmpeg-4.4.gz
+    sed -i 's/depends="download-javafx"//g' java/libraries/javafx/build.xml
+    sed -i '888,889d' java/libraries/javafx/src/processing/javafx/PSurfaceFX.java
+    cd build
+    ant build
+    cd ..
   '';
 
   installPhase = ''
     mkdir $out
-    cp -dpR build/linux/work $out/${pname}
-
+    cp -dpr build/linux/work $out/${pname}
     rmdir $out/${pname}/java
-    ln -s ${jdk} $out/${pname}/java
-
+    ln -s ${jdk11} $out/${pname}/java
     makeWrapper $out/${pname}/processing      $out/bin/processing \
-        --prefix XDG_DATA_DIRS : ${gsettings-desktop-schemas}/share/gsettings-schemas/${gsettings-desktop-schemas.name} \
-        --prefix _JAVA_OPTIONS " " -Dawt.useSystemAAFontSettings=lcd \
-        --prefix LD_LIBRARY_PATH : ${libXxf86vm}/lib
+      --prefix XDG_DATA_DIRS : ${gsettings-desktop-schemas}/share/gsettings-schemas/${gsettings-desktop-schemas.name} \
```

Comment:
use `wrapGAppsHook` and set `dontWrapGApps = true;` and add `''${gappsWrapperArgs[@]}` to here

Reply:
https://nixos.org/manual/nixpkgs/unstable/#ssec-gnome-hooks